pyvista/core/datasetattributes.py

```diff
@@ -750,8 +749,24 @@ def _prepare_array(
         """Prepare an array to be added to this dataset."""
         if data is None:
             raise TypeError('``data`` cannot be None.')
-        if isinstance(data, Iterable):
-            data = pyvista_ndarray(data)
+
+        # attempt to reuse the existing pointer to underlying VTK data
+        if isinstance(data, pyvista_ndarray):
+            # pyvista_ndarray already contains the reference to the vtk object
+            # pyvista needs to use the copy of this object rather than wrapping
+            # the array (which leaves a C++ pointer uncollected.
+            if data.VTKObject is not None:
+                # VTK doesn't support strides, therefore we can't directly
+                # point to the underlying object
+                if data.__array_interface__['strides'] is None:
+                    vtk_arr = copy_vtk_array(data.VTKObject, deep=deep_copy)
+                    if isinstance(name, str):
+                        vtk_arr.SetName(name)
+                        return vtk_arr
```

Comment:
If we're being careful with references, shouldn't we first check if the name is an str, and only call `copy_vtk_array()` when `isinstance(name, str)`? Or, alternatively, can we `return vtk_arr` when the name is unset?

Reply:
Opted to returning vtk_arr always and setting the name in a branch.

Resolved in 129477c3c4867152eb1dfce0f50730aa61c3bc0c.